### DISCUSSION TITLE
auto reload config on file modify

### DISCUSSION THREAD

**Discussion by iacore** [davmac314/dinit#284] [Q&A] [answered]
I want dinit to auto reload config file if it can be parsed.

Is this possible?

Is this a good idea?

**Comment by mobin-2008** [comment, marked as answer]
Hi.

I think it's not a good idea because a possible implementation for this is to create a "watcher" for each loaded service which is waste of resources (especially memory).

Also workaround for this is so simple, add this to your .bashrc (or any file which get loaded on your shell startup):
```sh
editsrv()
{
    ${EDITOR} "$1" # Or your fav text editor
    dinitcheck "$1" || return 1 # Check it for errors
    dinitctl reload "$1"
}
```

So any time you need to modify a service and reload it right after that, just use that function:
```sh
sudo editsrv dbus
editsrv pipewire
```

**Reply by iacore** [reply]
i found a bug in dinitcheck! it doesn't substitute environment variable

**Comment by davmac314** [comment]
I do not much like the idea of auto-reloading service descriptions. As a general rule, Dinit avoids implicit behaviour. I have been thinking it might be good to warn though if a service is started after the service description has changed.
